core/src/mindustry/input/DesktopInput.java

```diff
@@ -68,14 +68,28 @@ public void buildUI(Group group){
             t.touchable(() -> t.color.a < 0.1f ? Touchable.disabled : Touchable.childrenOnly);
             t.table(Styles.black6, b -> {
                 b.defaults().left();
-                b.label(() -> Core.bundle.format(!isBuilding ?  "resumebuilding" : "pausebuilding", Core.keybinds.get(Binding.pause_building).key.toString())).style(Styles.outlineLabel);
+                b.label(() -> Core.bundle.format(isBuilding ? "pausebuilding" : "resumebuilding", Core.keybinds.get(Binding.pause_building).key.toString())).style(Styles.outlineLabel);
                 b.row();
                 b.label(() -> Core.bundle.format("cancelbuilding", Core.keybinds.get(Binding.clear_building).key.toString())).style(Styles.outlineLabel);
                 b.row();
                 b.label(() -> Core.bundle.format("selectschematic", Core.keybinds.get(Binding.schematic_select).key.toString())).style(Styles.outlineLabel);
             }).margin(10f);
         });
 
+        //build enable hint
+        group.fill(t -> {
+            t.bottom();
```

Comment:
You should probably initialize this table with `t.color.a = 0` so it starts out hidden.

Reply:
Then shouldn't that also be applied to the general building hint table above?

Reply:
The difference is that this one starts out visible, while the other hints dialog does not.

Reply:
Testing without `t.color.a = 0`, that line doesn't seem to be necessary. Tested these scenarios:

* Starting fresh custom game with auto-pause off: build enable hint not visible until pressing E.
* Resuming saved game with building disabled: build enable hint visible as expected, disappears when pressing E.
* Resuming saved game with building enabled: build enable hint not visible as expected, appears when pressing E.
* Changing auto-pause setting during game: some anomalous behavior, like having to press E twice to get hint to appear.

[`isBuilding` starts out `true`](https://github.com/Anuken/Mindustry/blob/0cc9b0b0a3b7a30ae1dab30d3f194ac04f7152fc/core/src/mindustry/input/InputHandler.java#L60), which means the build enable hint is invisible at the start of the game.

Reply:
Fixed inconsistencies by replacing `!buildWasAutoPaused` with `!Core.settings.getBool("buildautopause")`. Now the build enable hint always reflects that state of `isBuilding` accurately, when auto-pause is disabled.